### PR TITLE
Report components docs

### PR DESCRIPTION
**Pull request by Aljullu**
In a private conversation with @timmyc via Slack we thought it would be a good idea to extend the docs to include the report components. They are used in several places and were growing in complexity, specially `<ReportTable>`, but there wasn't any documentation for them.

It made sense to use the same system of inline documentation we have for package components, instead of creating a new one for the report components. That required refactoring a bit the `bin/generate-docs` scripts.

### Detailed test instructions:
- Run `npm docs` and `npx docsify serve docs` and verify docs are created without errors.
- Read the new docs and suggest any improvement/correction you think makes sense.

All Isotopes are welcome to review this, but I'm tagging @ryelle because she worked in most of the docs until now.

### CODE REVIEW

File: bin/generate-docs/index.js [hunk 1]
```diff
@@ -12,45 +12,82 @@ const { parse, resolver } = require( 'react-docgen' );
...
 const {
+	ANALYTICS_FOLDER,
 	PACKAGES_FOLDER,
+	DOCS_FOLDER,
 	deleteExistingDocs,
 	getExportedFileList,
 	getMdFileName,
 	getRealFilePaths,
-	writeTableOfContents,
+	getTocContents,
```
Comment (coderkevin) [inline bin/generate-docs/index.js]: Nitpick: TOC Contents is redundant, just sayin'. 😉

File: bin/generate-docs/lib/file-system.js [hunk 2]
```diff
@@ -145,25 +150,32 @@ function isFile( file ) {
...
- * @param { array } files A list of readme files.
+ * @param { array } files A list of files, presumably in the components directory.
+ * @param { string } route Folder where the docs are stored.
+ * @param { string } title Title of the TOC section
+ * @return { string } TOC contents.
  */
-function writeTableOfContents( files ) {
-	const mdFiles = files.map( f => getMdFileName( f, false ) ).sort();
-	const TOC = uniq( mdFiles ).map( doc => {
+function getTocContents( files, route, title ) {
```
Comment (coderkevin) [inline bin/generate-docs/lib/file-system.js]: I like this, splitting up the generation of the TOC and the writing of the file.

File: client/analytics/components/report-summary/index.js [hunk 3]
```diff
@@ -70,10 +73,27 @@ export class ReportSummary extends Component {
 }
 
 ReportSummary.propTypes = {
+	/**
+	 * Properties of all the charts available for that report.
+	 */
 	charts: PropTypes.array.isRequired,
+	/**
+	 * The endpoint to use in API calls.
```
Comment (coderkevin) [inline client/analytics/components/report-summary/index.js]: I wonder if more description would be useful here. What does "endpoint" mean in this context. Maybe provide an example?

File: docs/components/analytics/report-summary.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+Properties of all the charts available for that report.
+
+### `endpoint`
+
+- **Required**
+- Type: String
+- Default: null
+
+The endpoint to use in API calls.
```
Comment (coderkevin) [inline docs/components/analytics/report-summary.md]: Again, maybe an example?

File: docs/components/analytics/report-table.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+- Default: null
+
+The key for user preferences settings for column visibility.
+
+### `endpoint`
+
+- Type: String
+- Default: null
+
+The endpoint to use in API calls.
```
Comment (coderkevin) [inline docs/components/analytics/report-table.md]: Same thing, an example would be nice.

**Comment by timmyc** [comment]
After the merge conflicts are addressed, I say we land this as-is and we can circle back and add in some more examples where @coderkevin said they might be nice to have. Awesome stuff @Aljullu !!

File: bin/generate-docs/index.js (hunk 1 above)
Comment (Aljullu) [reply bin/generate-docs/index.js]: Good point! Renamed it to `getTocSection`: ab8e53640eb0ba14865c9de59627568bcd279973.

File: client/analytics/components/report-summary/index.js (hunk 3 above)
Comment (Aljullu) [reply client/analytics/components/report-summary/index.js]: Done! :slightly_smiling_face: 

a1df848a2b944c1985e5ac96869b1abe2d2662fb

**Comment by Aljullu** [comment]
Merging as is per @timmyc's comment above.

**Pull request merged**